js/computation_graph.js

```diff
@@ -14,7 +14,10 @@ class ComputeNode {
         this.name = name;
         this.update_task = null;
         this.fail_cb = false;   // Set to true to force updates even if parent failed.
-        this.dirty = true;
+        this.dirty = 2;         // 3 states:
+                                // 2: dirty
+                                // 1: possibly dirty
+                                // 0: clean
```

Comment:
not 100% necessary but does js has enums

worth using them over int values?

Reply:
js does have enums*

Reply:
sad